### DISCUSSION TITLE
Quarto extension and Github subdirectory?

### DISCUSSION THREAD

**Discussion by mcanouil** [quarto-dev/quarto-cli#1440] [Q&A]
Hi,

does Quarto extensions/formats have to be alone in a Github repository?
I wanted to store several formats in one repository (https://github.com/mcanouil/quarto-ext) and apparently this is not currently possible.

```bash
quarto install extension mcanouil/quarto-ext/coeos
# Quarto extensions may execute code when documents are rendered. If you do not 
# trust the authors of the extension, we recommend that you do not install or 
# use the extension.
#  ? Do you trust the authors of this extension (Y/n) › Yes
# ERROR: TypeError: Invalid URL
```

Did I miss something when we want a subdirectory?

**Comment by jjallaire** [comment]
@dragonstyle Are we supposed to be able to install extensions from sub-directories?

**Reply by dragonstyle** [reply]
Right now, no - subdirectories are not supported. You can place multiple extensions in the `_extensions` directory of your repository and they will _all_ be installed when the root directory is a target for installation. They are not currently, individually targetable, however...

**Reply by mcanouil** [reply]
Is it something that could be considered in the future, i.e., as `remotes`, `renv` or `pak` for R packages?
(I actually do not mind the one extension = one repository strategy, especially since it makes things possibly clearer)

**Reply by dragonstyle** [reply]
Definitely- we should either support subdirectories or perhaps allow targeting / selection of specific extensions within the same repo. Seems useful!

**Reply by mcanouil** [reply]
Then, I'll let you create the issue from this discussion 😉

**Reply by mcanouil** [reply]
FYI, I went (for now) for the "one extension = one repository" and used git submodules to list them all in one repository https://github.com/mcanouil/quarto-ext
